### DISCUSSION TITLE
Unexpected result when I use gulp.dest()

### DISCUSSION THREAD

**Discussion by zhangzhiqiang37** [gulpjs/gulp#2504] [Help] [answered]
### What were you expecting to happen?
Hi, I have a problem with gulp.dest when I use it to double generate.
e.g.
```js
const stream = gulp.src(['./test.txt');
stream.pipe(gulp.dest('./zh'));
stream.pipe(gulp.dest('./en'));
```
I want to generate Chinese copy and English copy.

### What actually happened?

But I only get the English copy.
If I write the code in this way:
```js
gulp.src(['./test.txt');
  .pipe(gulp.dest('./zh'));
  .pipe(gulp.dest('./en'));
```
it works, I would get the right result, a Chinese copy and a English copy.

### Please give us a sample of your gulpfile

Here is github action https://github.com/zhangzhiqiang37/gulp-dest-test/runs/1281077531?check_suite_focus=true

### Please provide the following information:
* OS & version : MacOS Catalina 10.15.7
* node version : 8.11.3
* npm version : 5.6.0
* gulp version : 4.0.2

**Comment by phated** [comment, marked as answer]
That is actually just the behavior of nodejs streams. You can't pipe the result to 2 different outputs and you'd need something like a "tee stream", such as something like https://github.com/streamxorg/teex (I'm not sure if it works on standard node streams).

However, it's not recommended to split a gulp stream and instead apply your transforms inline. You can even filter files after they've been modified or written.

I'm going to convert this to a discussion because it is not really an issue with gulp.

**Reply by zhangzhiqiang37** [reply]
Why I can pipe the result to 2 different outputs when I use gulp@3.9.1, I'm puzzled about it

I'm updating gulp version from 3 to 4 in my project, they are different

See https://github.com/zhangzhiqiang37/gulp-dest-test/runs/1284541416?check_suite_focus=true (gulp@3.9.1 gulp dest to 2 different outputs test case)

**Reply by phated** [reply]
Gulp 3.9 streams weren't node compliant streams. It is one of the reasons for a major breaking version which you are opting to upgrade into.
